### PR TITLE
Extended functionality in EnsembleParameters

### PR DESCRIPTION
**Pull request by sigurdp**
* Extended functionality in EnsembleParameters class
* Introduced MinMax class for storing a min/max range

### CODE REVIEW

File: frontend/src/framework/EnsembleParameters.ts [hunk 1]
```diff
@@ -1,23 +1,124 @@
...
     constructor(parameterArr: Parameter[]) {
         this._parameterArr = parameterArr;
     }
 
-    getParameterNames(): string[] {
-        return this._parameterArr.map((par) => par.name);
+    getParameterIdents(requiredParamType: ParameterType | null): ParameterIdent[] {
+        const identArr: ParameterIdent[] = [];
+        for (const par of this._parameterArr) {
+            if (requiredParamType == null || par.type === requiredParamType) {
```
Comment (jorgenherje) [inline frontend/src/framework/EnsembleParameters.ts:85]: Should it be `===` for `requiredParamType == null`?

File: frontend/src/framework/EnsembleParameters.ts [hunk 2]
```diff
@@ -1,23 +1,124 @@
...
+                identArr.push(new ParameterIdent(par.name, par.groupName));
+            }
+        }
+
+        return identArr;
+    }
+
+    hasParameter(paramIdent: ParameterIdent): boolean {
+        return this.findParameter(paramIdent) != null;
```
Comment (jorgenherje) [inline frontend/src/framework/EnsembleParameters.ts:94]: Should it be `!==`?

File: frontend/tests/unit-tests/EnsembleParameters.test.ts [hunk 3]
```diff
@@ -0,0 +1,118 @@
...
+        {
+            const minMax = ensParams.getContinuousParameterMinMax(ParameterIdent.fromNameAndGroup("cparam_50", "grp1"));
+            expect(minMax).toEqual(new MinMax(51, 54));
+        }
+    });
+});
+
+
+describe("ParameterIdent tests", () => {
+    test("Conversion to/from string", () => {
```
Comment (jorgenherje) [inline frontend/tests/unit-tests/EnsembleParameters.test.ts:103]: Perhaps a matter of taste, but I like to test single functions per test, using hard coded string as expected result for string conversion and hard coded strings as argument for generators from string (`fromString()`method). I.e. test `fromNameAndGroup("aName", "aGroup")`-method towards a `new ParameterIdent("aName", "aGroup")`-object and its public properties? Then test `toString()`-method towards a hard coded expected string? Then testing `fromString()` with the hardcoded string?

**Review by jorgenherje — CHANGES_REQUESTED** [review]
Looks good to me. Nice functionality!

Minor comments

File: frontend/src/framework/EnsembleParameters.ts [hunk 4]
```diff
@@ -1,23 +1,124 @@
...
+
+    static fromString(paramIdentString: string): ParameterIdent {
+        const parts = paramIdentString.split("~@@~");
+        if (parts.length === 1) {
+            return new ParameterIdent(parts[0], null);
+        } else if (parts.length === 2) {
+            return new ParameterIdent(parts[0], parts[1]);
+        } else {
+            throw new Error(`Invalid parameter ident string: ${paramIdentString}`);
+        }
```
[the remark this answers is not in the document]
Comment (sigurdp) [reply frontend/src/framework/EnsembleParameters.ts:52]: Agreed 👍

File: frontend/src/framework/EnsembleParameters.ts (hunk 1 above)
Comment (sigurdp) [reply frontend/src/framework/EnsembleParameters.ts:85]: The use of == was intentional and will catch undefined as well. Not a feature win but I see no harm.

File: frontend/src/framework/EnsembleParameters.ts (hunk 2 above)
Comment (sigurdp) [reply frontend/src/framework/EnsembleParameters.ts:94]: Doesn't make much difference, but will change it.

File: frontend/tests/unit-tests/EnsembleParameters.test.ts (hunk 3 above)
Comment (sigurdp) [reply frontend/tests/unit-tests/EnsembleParameters.test.ts:103]: Will make some adjustments to the test

**Review by jorgenherje — APPROVED** [review]
👍

File: frontend/src/framework/EnsembleParameters.ts (hunk 1 above)
Comment (jorgenherje) [reply frontend/src/framework/EnsembleParameters.ts:86]: 👍

**Pull request merged**